test/runtests.jl

```diff
@@ -122,6 +122,34 @@ end
     @test ShortString7(ShortString7("ab")) isa ShortString7
 
     @test_throws ErrorException ShortString3(ShortString7("123456"))
+
+    # Test creating shortstrings with maxlen
+    @test ShortString(
+        ss127"Be honest, do you actually need a string longer than this. Seriously. C'mon this is pretty long.",
+        127,
+    ) == "Be honest, do you actually need a string longer than this. Seriously. C'mon this is pretty long."
+    @test ShortString(
+        ss63"Basically a fairly long string really", 63
+    ) == "Basically a fairly long string really"
+    @test ShortString(ss31"A Longer String!!!", 31) == "A Longer String!!!"
+    @test ShortString(ss15"Short String!!!", 15) == "Short String!!!"
+    @test ShortString(ss7"ShrtStr", 7) == "ShrtStr"
+    @test ShortString(ss3"ss3", 3) == "ss3"
+    @test ShortString("", 0) == ""
+
+    @test_throws ErrorException ShortString(
+        ss127"Be honest, do you actually need a string longer than this. Seriously. C'mon this is pretty long.",
+        0,
+    ) == "Be honest, do you actually need a string longer than this. Seriously. C'mon this is pretty long."
+    @test_throws ErrorException ShortString(
+        ss63"Basically a fairly long string really", 0
+    ) == "Basically a fairly long string really"
+    @test_throws ErrorException ShortString(
+        ss31"A Longer String!!!", 0
+    ) == "A Longer String!!!"
+    @test_throws ErrorException ShortString(ss15"Short String!!!", 0) == "Short String!!!"
+    @test_throws ErrorException ShortString(ss7"ShrtStr", 0) == "ShrtStr"
```

Comment:
Including the right-hand side == here seems like it is adding nothing?
It makes it unclear what should throw the error.